components/encoder/incremental/incremental_encoder.go

```diff
@@ -190,12 +190,6 @@ func (e *Encoder) Start(ctx context.Context) {
 		defer e.A.RemoveCallback(chanA)
 		defer e.B.RemoveCallback(chanB)
 		for {
-			select {
-			case <-e.CancelCtx.Done():
-				return
-			default:
-			}
-
```

Comment:
I don't think I get how CancelCtx works enough to say if this is fine or not? This is just how it was already implemented it became a component

Reply:
Look down 5ish lines. It used to be basically:
```
if there's something in the Done channel, return
declare a new variable
if there's something in the Done channel, return
```
We don't need both of those.

Reply:
ah yes makes sense thank you!

Reply:
...oh! Actually you're totally right! This is really subtle, but if the select statement below can choose between multiple cases, it picks one at random. We want the Done channel to take precedence over the others, hence checking it separately up here. By removing this part, I open up the possibility that we never notice that we're supposed to return because there's too much data coming in from the encoder and we never actually check the Done channel. I'll put this back and add in a comment. Thanks for making me rethink this!